manipvortex/manipulator.go

```diff
@@ -697,41 +697,57 @@ func (m *vortexManipulator) monitor(ctx context.Context) {
 
 func (m *vortexManipulator) pushEvent(evt *elemental.Event) {
 
+	m.RLock()
+	defer m.RUnlock()
+
 	for _, s := range m.subscribers {
 		sevent, err := copystructure.Copy(evt)
 		if err != nil {
 			zap.L().Error("failed to copy event", zap.Error(err))
 			continue
 		}
 
-		if !s.filter.IsFilteredOut(evt.Identity, evt.Type) {
+		s.RLock()
+		if !s.filter.IsFilteredOut(evt.Identity, evt.Type) && evt.Type == elemental.EventDelete && evt.Identity == "namespace" {
 			select {
 			case s.subscriberEventChannel <- sevent.(*elemental.Event):
 			default:
-				zap.L().Error("Subscriber channel is full")
+				zap.L().Debug("Dropping delete event as channel is full")
 			}
 		}
+		s.RUnlock()
 	}
 }
 
 func (m *vortexManipulator) pushStatus(status manipulate.SubscriberStatus) {
 
+	m.RLock()
+	defer m.RUnlock()
+
 	for _, s := range m.subscribers {
 		select {
 		case s.subscriberStatusChannel <- status:
 		default:
-			zap.L().Error("Subscriber channel is full")
+			zap.L().Error("Subscriber status channel is full")
 		}
 	}
 }
 
 func (m *vortexManipulator) pushErrors(err error) {
+
+	m.RLock()
+	defer m.RUnlock()
+
 	for _, s := range m.subscribers {
+		s.Lock()
+
 		select {
 		case s.subscriberErrorChannel <- err:
 		default:
-			zap.L().Error("Subscriber channel is full")
+			zap.L().Error("Subscriber error channel is full")
 		}
+
+		s.Unlock()
```

Comment:
Why do you need to lock the subscriber? This lock only protects the filter and it is not a problem if it changes.

Reply:
Yeah, not sure why I did this one and I cannot find a prior data race. Let me remove.